src/demandlib/vdi/regions.py

```diff
@@ -0,0 +1,598 @@
+
+        Parameters
+        ----------
+        year : int
+            Year of the profile.
+        try_region : int
+            The number of the TRY (test reference year of the DWD) region.
+        seasons : dict (optional)
+            The times of the seasons if fixed seasons are used. In the VDI
+            norm seasons are defined by the daily average temperature: Winter
+            below 5 degree Celsius, Spring between 5 and 15 degree Celsius and
+            summer above 15 degree Celsius.
+        holidays : dict or list
+            In case of a dictionary the keys are datetime objects and the
+            values are strings with the name of the holiday. Otherwise a list
+            of datetime objects should be passed.
+        houses : list
+            A list of dictionaries in which each house is defined the
+            dictionary need to have the following keys.
+        resample_rule : str
+            Time interval to resample the profile e.g. 1H (1 hour) or 15min.
+            The value will be passed to the pandas resample method.
+        """
+        if calendar.isleap(year):
+            self.hoy = 8784
+        else:
+            self.hoy = 8760
+        self._datapath = os.path.join(os.path.dirname(__file__), "bdew_data")
+
+        self._seasons = {
+            "summer1": [5, 15, 9, 14],  # summer: 15.05. to 14.09
+            "transition1": [3, 21, 5, 14],  # transition1 :21.03. to 14.05
+            "transition2": [9, 15, 10, 31],  # transition2 :15.09. to 31.10
+            "winter1": [1, 1, 3, 20],  # winter1:  01.01. to 20.03
+            "winter2": [11, 1, 12, 31],  # winter2: 01.11. to 31.12
+        }
+
+        if seasons is not None:
+            self._seasons.update(seasons)
+
+        self._try_region = try_region
+
+        self._year = year
+        self.weather = None
+        self.houses = []
+        if houses is not None:
+            self.add_houses(houses)
+
+        self.temperature_limits = self._get_temperature_level_combinations()
+        self.type_days = {}
+        self._load_profiles = {}
+        for temp_limit in self.temperature_limits:
+            self.type_days[temp_limit] = self._get_typical_days(
+                holidays, temp_limit
+            )
+            self._load_profiles[temp_limit] = self._load_profile_factors(
+                temp_limit
+            )
+
+            if resample_rule is not None:
+                self._resample_profiles(resample_rule, temp_limit)
+
+    def _resample_profiles(self, rule, tl):
+        self._load_profiles[tl] = self._load_profiles[tl].resample(rule).sum()
+        self.type_days[tl] = self.type_days[tl].resample(rule).first()
+
+    def _get_temperature_level_combinations(self):
+        t_limit = namedtuple("temperature_limit", "summer winter")
+        return set(
+            [
+                t_limit(
+                    summer=h["summer_temperature_limit"],
+                    winter=h["winter_temperature_limit"],
```

Comment:
Do not fail if the temperature options are not defined. Use the default values instead:
```suggestion
                    summer=h.get("summer_temperature_limit", 15),
                    winter=h.get("winter_temperature_limit", 5),
```